packer/build.go

```diff
@@ -352,7 +352,7 @@ PostProcessorRunSeqLoop:
 		copy(artifacts[1:], artifacts)
 		artifacts[0] = builderArtifact
 	} else {
-		log.Printf("Deleting original artifact for build '%s'", b.name)
+		log.Printf("Deleting original artifact for build '%s'", b.Type)
```

Comment:
Yes ! I wanted to comment about this; so the builder interface has a `Name()` func ( so this name hard to change ) but I needed to make `name` public in order to be able to create a builder from another package. Because in **most** cases (because it is the default) when we build something the type of the builder is used here (eg: `virtualbox-iso` or `amazon-ebs`): changing this to `Type` was the simplest/fastest way I could find to do this. 

Do you have alternative ideas ? 🤔

Reply:
May be I could name it `Identifier` ?

Reply:
I'm just imagining the situation where we have two amazon-ebs builders, one named "ebs-1" and one named "ebs-2" -- will it print `"Deleting original artifact for build 'amazon-ebs'` for both, now, or will it print `"Deleting original artifact for build 'ebs-1'` and `"Deleting original artifact for build ebs-2'`?  Type is already very specifically used, so on a casual reading I would assume the first (amazon-ebs for both)

Reply:
For now it will show something like this :

<img width="210" alt="Screenshot 2019-12-05 at 16 20 16" src="https://user-images.githubusercontent.com/1024852/70248418-28bb4800-177b-11ea-938a-351ff53c1bc5.png">

It is not yet possible to name a build with HCL2. In the future I want to be able to add the `build_from` block that will replace the `from_sources` list.
```hcl
from_sources = [
        "src.amazon-ebs.first",
        "src.amazon-ebs.first",
]
```
So that we can name things there.

```hcl
build_from "src.amazon-ebs.first" "name" {
    // overload specific things here
}
```

Reply:
To enable this; today I changed this :
https://github.com/hashicorp/packer/blob/5ff562343326f76b63f0b07d52831385e6b143b8/command/build.go#L144
to this: 

https://github.com/hashicorp/packer/blob/9eb3ecb1248c3433833cd8056620c3375055abeb/command/build.go#L196

Reply:
We don't have to do it in this PR but I do think it'll be necessary to do it before we consider HCL2 supported.

Reply:
Got it ! Yes this will be a great addition.